mlir/unittests/Interfaces/DataLayoutInterfacesTest.cpp

```diff
@@ -131,6 +138,8 @@ struct TypeNoLayout : public Type::TypeBase<TypeNoLayout, Type, TypeStorage> {
 
   using Base::Base;
 
+  static constexpr StringRef getTypeName() { return "test.no_layout"; }
+
```

Comment:
One thing with these accessors is that they will interfere with the auto-generated accessors for the parameters, what happens in this case?
Should ODS forbid a parameter named `name`?